### PR TITLE
adapts sph-diagram-generation to AutoPas interface

### LINKED ISSUE #273: sph-diagram-generation should be using AutoPas interface
**Describe the bug**
Currently does not use AutoPas interface and thus ignores rebuild frequency.

**Expected behavior**
Uses AutoPas interface and does not ignore rebuild frequency

### PR DESCRIPTION
**Pull request by SteffenSeckler**
# Description

Adapts sph-diagram-generation to AutoPas interface and adapts measurePerf.sh script accordingly

## Resolved Issues

- [x] fixes #273 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] locally run the measurePerf.sh script -- produced no errors and proper results.

### CODE REVIEW

File: examples/sphDiagramGeneration/sph-diagram-generation.cpp [hunk 1]
```diff
@@ -132,82 +100,24 @@ int main(int argc, char *argv[]) {
...
-template <class Container, class Functor>
-void measureContainer(Container *cont, Functor *func, int numParticles, int numIterations, bool useNewton3) {
-  using CellType = autopas::FullParticleCell<autopas::sph::SPHParticle>;
-  // initialize to dummy values
-  auto traversalInfo = cont->getTraversalSelectorInfo();
-  std::cout << "Cells: " << traversalInfo.dims[0] << " x " << traversalInfo.dims[1] << " x " << traversalInfo.dims[2]
-            << std::endl;
   auto traversalType = autopas::TraversalOption(-1);
-  switch (cont->getContainerType()) {
+  switch (*containerOptions.begin()) {
```
Comment (FG-TUM) [inline examples/sphDiagramGeneration/sph-diagram-generation.cpp]: I don't understand this `switch`. Why does it only check the first container option? Why do you only allow one traversal per container and why not let the user decide which one?

File: examples/sphDiagramGeneration/sph-diagram-generation.cpp [hunk 2]
```diff
@@ -60,47 +55,30 @@ int main(int argc, char *argv[]) {
...
-  } else if (argc == 8) {
-    numParticles = std::stoi(argv[1]);
-    numIterations = std::stoi(argv[2]);
-    containerTypeInt = std::stoi(argv[3]);
-    functorTypeInt = std::stoi(argv[4]);
-    skin = std::stod(argv[5]);
-    rebuildFrequency = std::stoi(argv[6]);
+  }
+  if (argc >= 8) {
     useNewton3 = std::stoi(argv[7]);
```
Comment (FG-TUM) [inline examples/sphDiagramGeneration/sph-diagram-generation.cpp]: you are filling a bool with `stoi`. Just saying...

Comment (SteffenSeckler) [reply examples/sphDiagramGeneration/sph-diagram-generation.cpp]: the only proper alternative is to use boost::lexical_cast, which I don't want to ;)
or writing your own proper parser, which I also don't want to do.

File: examples/sphDiagramGeneration/sph-diagram-generation.cpp (hunk 1 above)
Comment (SteffenSeckler) [reply examples/sphDiagramGeneration/sph-diagram-generation.cpp]: only 1 container option is allowed ;)

File: examples/sphDiagramGeneration/sph-diagram-generation.cpp (hunk 2 above)
Comment (SteffenSeckler) [reply examples/sphDiagramGeneration/sph-diagram-generation.cpp]: and currently we only allow one traversal

File: examples/sphDiagramGeneration/sph-diagram-generation.cpp (hunk 1 above)
Comment (FG-TUM) [reply examples/sphDiagramGeneration/sph-diagram-generation.cpp]: ok didn't see the if in line 76.

**Pull request merged**